python/private/stage2_bootstrap_template.py

```diff
@@ -468,68 +384,17 @@ def main():
-        if cov_tool is None:
-            print_verbose_coverage(
-                "Coverage was enabled, but python coverage tool was not configured."
-                + "To enable coverage, consult the docs at "
-                + "https://rules-python.readthedocs.io/en/latest/coverage.html"
-            )
-        else:
-            # Inhibit infinite recursion:
-            if "PYTHON_COVERAGE" in os.environ:
-                del os.environ["PYTHON_COVERAGE"]
-
-            if not os.path.exists(cov_tool):
-                raise EnvironmentError(
-                    "Python coverage tool %r not found. "
-                    "Try running with VERBOSE_COVERAGE=1 to collect more information."
-                    % cov_tool
-                )
-
-            # coverage library expects sys.path[0] to contain the library, and replaces
-            # it with the directory of the program it starts. Our actual sys.path[0] is
-            # the runfiles directory, which must not be replaced.
-            # CoverageScript.do_execute() undoes this sys.path[0] setting.
-            #
-            # Update sys.path such that python finds the coverage package. The coverage
-            # entry point is coverage.coverage_main, so we need to do twice the dirname.
-            coverage_dir = os.path.dirname(os.path.dirname(cov_tool))
-            print_verbose("coverage: adding to sys.path:", coverage_dir)
-            python_path_entries.append(coverage_dir)
-            python_path_entries = deduplicate(python_path_entries)
-    else:
-        cov_tool = None
-
     sys.stdout.flush()
 
-    # Add the user imports after the stdlib, but before the runtime's
-    # site-packages directory. This gives the stdlib precedence, while allowing
-    # users to override non-stdlib packages that may have been bundled with
-    # the runtime (usually pip).
-    # NOTE: There isn't a good way to identify the stdlib paths, so we just
-    # expect site-packages comes after it, per
-    # https://docs.python.org/3/library/sys_path_init.html#sys-path-init
-    for i, path in enumerate(sys.path):
-        # dist-packages is a debian convention, see
-        # https://wiki.debian.org/Python#Deviations_from_upstream
-        if os.path.basename(path) in ("site-packages", "dist-packages"):
-            sys.path[i:i] = python_path_entries
-            break
-    else:
-        # Otherwise, no site-packages directory was found, which is odd but ok.
-        sys.path.extend(python_path_entries)
-
     # NOTE: The sys.path must be modified before coverage is imported/activated
     # NOTE: Perform this after the user imports are appended. This avoids a
     # user import accidentally triggering the site-packages logic above.
     sys.path[0:0] = prepend_path_entries
 
-    with _maybe_collect_coverage(enable=cov_tool is not None):
+    # todo: also check COVERAGE_DIR env var
+    import _bazel_site_init
+
+    with _maybe_collect_coverage(enable=_bazel_site_init.COVERAGE_SETUP):
```

Comment:
Nice that we can import things from the `_bazel_site_init`, nit, please create a ticket for the TODO above unless you plan to work on this soon. :)

Reply:
Addressed.